### PR TITLE
[JSON-API] Correctly extract the request source URL/IP

### PR DESCRIPTION
**Pull request by realvictorprm**
changelog_begin

- [JSON-API] If the service is put behind a proxy filling either of these headers X-Forwarded-For & X-Real-Ip then these will now be respected for logging the request source ip/url

changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala [hunk 1]
```diff
@@ -67,12 +68,23 @@ class Endpoints(
   import util.ErrorOps._
   import Uri.Path._
 
+  // Inspired by
```
Comment (cocreature) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala:71]: I assume the idea is to switch over to that once we use the dsl?

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/HttpService.scala [hunk 2]
```diff
@@ -195,16 +193,14 @@ object HttpService {
         websocketService,
       )
 
-      ignoreConParam = (res: Future[HttpResponse]) => (_: Http.IncomingConnection) => res
```
Comment (cocreature) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/HttpService.scala:198]: I don’t understand the changes here. How are they related?

Comment (realvictorprm) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/HttpService.scala:198]: The con param was only used for extracting the URL/IP.

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala (hunk 1 above)
Comment (realvictorprm) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala:71]: Correct

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/HttpService.scala (hunk 2 above)
Comment (cocreature) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/HttpService.scala:198]: ah makes sense, thanks for the explanation.

**Review by cocreature — APPROVED** [review]
Thanks!

**Review by S11001001 — APPROVED** [review]

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala [hunk 3]
```diff
@@ -147,17 +159,16 @@ class Endpoints(
...
-          } yield res
-        })
+      extendWithRequestIdLogCtx(implicit lc => {
+        val t0 = System.nanoTime
+        logger.info(s"Incoming request on ${req.uri} from ${requestSource(req)}")
+        metrics.daml.HttpJsonApi.httpRequestThroughput.mark()
+        for {
+          res <- lcFhr(lc)
+          _ = logger.trace(s"Processed request after ${System.nanoTime() - t0}ns")
+          _ = logger.info(s"Responding to client with HTTP ${res.status}")
```
[the remark this answers is not in the document]
Comment (realvictorprm) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala]: many thanks! :)

**Pull request merged**